src/lib.rs

```diff
@@ -149,6 +180,33 @@ impl From<&Opcode> for Vec<u8> {
     }
 }
 
+/// A Zcash script consists of a sig and a pubkey.
+///
+/// - Script<opcode::PossiblyBad, opcode::PossiblyBad> – from the chain
+/// - Script<opcode::PushValue, opcode::Structured> – authoring sig_push_only
+/// - Script<opcode::Structured, opcode::Structured> – authoring non-sig_push_only
+///
+/// Until `Structured` is added, we use `Opcode` in its place.
```

Comment:
It is not necessarily clear from this documentation that the first type parameter restricts the sig and the second restricts the pubkey, so that should be said explicitly.